src/tests/java/picard/analysis/CollectMultipleMetricsTest.java

```diff
@@ -0,0 +1,249 @@
+package picard.analysis;
+
+import htsjdk.samtools.metrics.MetricsFile;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import picard.cmdline.CommandLineProgramTest;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+/**
+ * Tests the two default "programs" that have tests in CollectMultipleMetrics
+ *
+ *
+ * @author Yossi farjoun
+ */
+
+public class CollectMultipleMetricsTest extends CommandLineProgramTest {
+
+    private static final File TEST_DATA_DIR = new File("testdata/picard/sam");
+
+    public String getCommandLineProgramName() {
+        return CollectMultipleMetrics.class.getSimpleName();
+    }
+
+
+    @Test
+    public void testAlignmentSummaryViaMultipleMetrics() throws IOException {
+        final File input = new File(TEST_DATA_DIR, "summary_alignment_stats_test.sam");
+        final File reference = new File(TEST_DATA_DIR, "summary_alignment_stats_test.fasta");
+        final File outfile   = File.createTempFile("alignmentMetrics", ".txt");
+        outfile.deleteOnExit();
+        final String[] args = new String[] {
+                "INPUT="  + input.getAbsolutePath(),
+                "OUTPUT=" + outfile.getAbsolutePath(),
+                "REFERENCE_SEQUENCE=" + reference.getAbsolutePath(),
+                "PROGRAM=null",
+                "PROGRAM="+CollectMultipleMetrics.Program.CollectAlignmentSummaryMetrics.name(),
+                "PROGRAM="+CollectMultipleMetrics.Program.CollectInsertSizeMetrics.name()
+        };
+        Assert.assertEquals(runPicardCommandLine(args), 0);
+
+        final MetricsFile<AlignmentSummaryMetrics, Comparable<?>> output = new MetricsFile<AlignmentSummaryMetrics, Comparable<?>>();
+        output.read(new FileReader(outfile));
+
+        for (final AlignmentSummaryMetrics metrics : output.getMetrics()) {
```

Comment:
We seem to be not looking at the same lines of code.

the lines of code you marked are a duplicate of what?

On Wed, Apr 8, 2015 at 1:39 PM, Nils Homer <notifications@github.com> wrote:

> In src/tests/java/picard/analysis/CollectMultipleMetricsTest.java
> <https://github.com/broadinstitute/picard/pull/195#discussion_r27993404>:
>
> > +        final File outfile   = File.createTempFile("alignmentMetrics", ".txt");
> > +        outfile.deleteOnExit();
> > +        final String[] args = new String[] {
> > +                "INPUT="  + input.getAbsolutePath(),
> > +                "OUTPUT=" + outfile.getAbsolutePath(),
> > +                "REFERENCE_SEQUENCE=" + reference.getAbsolutePath(),
> > +                "PROGRAM=null",
> > +                "PROGRAM="+CollectMultipleMetrics.Program.CollectAlignmentSummaryMetrics.name(),
> > +                "PROGRAM="+CollectMultipleMetrics.Program.CollectInsertSizeMetrics.name()
> > +        };
> > +        Assert.assertEquals(runPicardCommandLine(args), 0);
> > +
> > +        final MetricsFile<AlignmentSummaryMetrics, Comparable<?>> output = new MetricsFile<AlignmentSummaryMetrics, Comparable<?>>();
> > +        output.read(new FileReader(outfile));
> > +
> > +        for (final AlignmentSummaryMetrics metrics : output.getMetrics()) {
>
> I disagree. The asserts are the same, and the input and reference files
> are the same.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/picard/pull/195/files#r27993404>.
>